--- a/drivers/net/wireless/ath/ath12k/pci.c
+++ b/drivers/net/wireless/ath/ath12k/pci.c
@@ -774,7 +774,9 @@ static int ath12k_pci_msi_alloc(struct a
 
 	ath12k_pci_msi_disable(ab_pci);
 
-	msi_desc = irq_get_msi_desc(ab_pci->pdev->irq);
+	msi_desc = irq_get_msi_desc(pci_irq_vector(ab_pci->pdev, 0));
+	if (!msi_desc)
+		msi_desc = irq_get_msi_desc(ab_pci->pdev->irq);
 	if (!msi_desc) {
 		ath12k_err(ab, "msi_desc is NULL!\n");
 		ret = -EINVAL;
@@ -805,7 +807,9 @@ static int ath12k_pci_config_msi_data(st
 {
 	struct msi_desc *msi_desc;
 
-	msi_desc = irq_get_msi_desc(ab_pci->pdev->irq);
+	msi_desc = irq_get_msi_desc(pci_irq_vector(ab_pci->pdev, 0));
+	if (!msi_desc)
+		msi_desc = irq_get_msi_desc(ab_pci->pdev->irq);
 	if (!msi_desc) {
 		ath12k_err(ab_pci->ab, "msi_desc is NULL!\n");
 		pci_free_irq_vectors(ab_pci->pdev);
